cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java

```diff
@@ -80,7 +80,8 @@ public FactTable get(int resolution) {
     int ttl =  cConf.getInt(MetricsConstants.ConfigKeys.RETENTION_SECONDS + "." + resolution + ".seconds", -1);
 
     DatasetProperties.Builder props = DatasetProperties.builder();
-    if (ttl > 0) {
+    // don't add TTL for MAX_RESOLUTION table.
+    if (ttl > 0 && resolution != Integer.MAX_VALUE) {
```

Comment:
Is there an easy way to make that configurable? I see above lines try to get it from config. Can we default it in config to unlim or is it toodangerous?